### PR TITLE
chore: update cardinal version to v1.2.4-beta for cardinal editor

### PR DESCRIPTION
**Pull request by zulkhair**
Closes: WORLD-XXX

## Overview
Update version of cardinal to support cardinal editor

## Testing and Verifying
Manually run using `world cardinal dev` command

### CODE REVIEW

**Comment by zulkhair** [comment]
* **#41** <a href="https://app.graphite.dev/github/pr/Argus-Labs/starter-game-template/41?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @zulkhair and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cardinal/go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ replace (
 
 require (
 	github.com/rs/zerolog v1.31.0
-	pkg.world.dev/world-engine/cardinal v1.2.2-beta
+	pkg.world.dev/world-engine/cardinal v1.2.3-beta
```
Comment (rmrt1n) [inline cardinal/go.mod]: v1.2.3-beta is outdated, bump to v1.2.4

Comment (rmrt1n) [reply cardinal/go.mod]: note: cardinal editor v0.1.0 works up until cardinal v1.2.3-beta. the latest version of the editor (still untagged, v0.2.0) only works with cardinal v1.2.4-beta due to some breaking changes

Comment (zulkhair) [reply cardinal/go.mod]: ok, updated to the latest version

**Review by rmrt1n — APPROVED** [review]

**Pull request merged**